### DISCUSSION TITLE
Data validation with Zod and TypeScript

### DISCUSSION THREAD

**Discussion by Gpx** [TanStack/query#3366] [Show and tell]
Hi all, I started using the approach described in [this article](https://polvara.me/posts/effective-query-functions-for-react-query-with-zod) to validate my query request and response data with [Zod](https://zod.js.org/) and TypeScript. I'll post the interesting bit of code here:

```ts
const GetRepositoryRequest = z.void();
const GetRepositoryResponse = z.object({ name: z.string().min(1) });
export const getRepository = api<
  z.infer<typeof GetRepositoryRequest>,
  z.infer<typeof GetRepositoryResponse>
>({
  method: HTTPMethod.GET,
  path: "/repository",
  requestSchema: GetRepositoryRequest,
  responseSchema: GetRepositoryResponse,
});
```

I was wondering what everyone thought about it. If people find it useful, I'm thinking of publishing a library that generalizes the same approach.

Thanks for the feedback!

**Comment by olivierdelomez** [comment]
@Gpx I know it's an old thread but:

I went to use your approach and it's great, thanks for sharing it :)

The minor change I've made is to pass the `path` next to the `requestData` because I needed dynamic values in my `path`.. like so `getRepository(path, data)`

**Reply by Gpx** [reply]
I'm glad it helped @olivierdelomez 😄
